packages/common/src/utils/chatUtils.ts

```diff
@@ -45,3 +47,14 @@ export const isEarliestUnread = ({
   const isAuthor = message.sender_user_id === currentUserId
   return isUnread && !isPreviousMessageUnread && !isAuthor
 }
+
+/**
+ * Can only fetch more messages if it's not in a loading state and
+ * there are previous messages to fetch
+ */
+export const chatCanFetchMoreMessages = (
```

Comment:
very nit but this could even take a chatID and just be a selector?

Reply:
discussed, leaving as is